src/node/int.ts

```diff
@@ -0,0 +1,16 @@
+import * as assert from 'assert';
+
+import { Node } from './base';
+
+export class Int extends Node {
+  /**
+   * @param field  State's property name
+   */
+  constructor(public readonly field: string, public readonly bytes: number, public readonly signed: boolean, public readonly littleEndian: boolean) {
+    super(field + '_' + (signed ? 'int' : 'uint') + '_' + (bytes * 8) + (bytes > 1 ? '_' + (littleEndian ? 'le' : 'be') : ''));
```

Comment:
Nitpick: let's move compound expressions to separate variables and use template syntax here.

Reply:
That#s a bit problematic - the `super` call needs to be the very first instruciton in the constuctor. 😞I'll try to clean this up a but with the template syntax.

Reply:
I extracted the name generation to a separate function to work around the `super` call requirement. 👍

Reply:
Yay, looks great! :+1: